crates/ruff_python_formatter/src/expression/string.rs

```diff
@@ -639,3 +655,301 @@ fn normalize_string(
 
     (normalized, newlines)
 }
+
+/// For docstring indentation, black counts spaces as 1 and tabs by increasing the indentation up
+/// to the next multiple of 8. This is effectively a port of `str.expandtabs`, which black calls.
+fn count_indentation_like_black(line: &str) -> TextSize {
+    let tab_width: u32 = 8;
+    let mut indentation = TextSize::default();
+    for char in line.chars() {
+        if char == '\t' {
+            // Pad to the next multiple of tab_width
+            indentation += TextSize::from(tab_width - (indentation.to_u32().rem_euclid(tab_width)));
+        } else if char.is_whitespace() {
+            indentation += char.text_len();
+        } else {
+            return indentation;
+        }
+    }
+    indentation
+}
+
+/// Format a docstring by trimming whitespace and adjusting the indentation.
+///
+/// Summary of changes we make:
+/// * Normalize the string like all other strings
+/// * Ignore docstring that have an escaped newline
+/// * Trim all trailing whitespace, except for a chaperone space to avoid quotes or backslashes
+///   in the last line.
+/// * Trim leading whitespace on the first line, again except for a chaperone space
```

Comment:
```suggestion
/// * Trim leading whitespace on the first line, again except for a chaperone space
```

What's a chaperone space? Googling doesn't provide any good results

Reply:
i made this term up from a [similar one](https://en.wikipedia.org/wiki/Chaperone_(protein): it's a space that makes sure that quote or backspace inside the string don't get to close the surrounding quotes and start misbehaving

Reply:
Oh okay. We should define it somewhere (if it isn't already the case)